src/sphinx_autodoc_typehints/__init__.py

```diff
@@ -609,7 +609,11 @@ def _inject_types_to_docstring(
 
                 lines.insert(insert_index, type_annotation)
 
-    if "return" in type_hints and not inspect.isclass(original_obj):
+    if (
+        "return" in type_hints
+        and not inspect.isclass(original_obj)
+        and not inspect.isdatadescriptor(original_obj)  # noqa: SC200 # Possibly misspelt word: 'isdatadescriptor'
```

Comment:
Add this word to whitelist.txt instead.

Reply:
Okay. We also need to add `getitem` there...